virtual_rainforest/core/model.py

```diff
@@ -1,61 +1,69 @@
-``virtual_rainforest`` model.
-
-However, a more complex form of inheritance is required for functions that define the
-shared api. These should take the same input and perform an equivalent set of steps
-across models, but because they interact with radically different modules their internal
-workings will have little in common. Thus,
-:class:`~virtual_rainforest.core.model.BaseModel` is defined as an abstract base class,
-which is a class that is never intended to be instantiated itself but instead serves as
-a blueprint for inheriting classes. This type of class can define abstract methods
-(denoted by ``@abstractmethod``), which are merely placeholders. For these abstract
-methods, child classes have to overwrite the methods with new functions, otherwise class
-inheritance fails. This ensures that a consistent api (set of functions) is used across
-models, while also ensuring that functions don't default to an inappropriate generic
-behaviour due to a function not being defined for a particular model. At the moment, we
-expect every model to have a setup, spinup, update and cleanup process, though this
-might change in the future.
-
-We also define an abstract class method to perform model initialisation. This method
-(:func:`~virtual_rainforest.core.model.BaseModel.from_config`) is a factory method which
-unpacks the configuration dictionary, extracts the relevant tags and attempts to convert
-them into the form needed to initialise a class instance. This method must be defined
-for every child class of :class:`~virtual_rainforest.core.model.BaseModel`, as unpacking
-our complex configuration dictionary is a necessary before a class instance can be
-initialised.
-
-As in the case of configuration schema, we make ``Model`` classes generally accessible
-by adding them to a registry. However, in this case the function to add to the registry
-isn't a decorator but rather a member function of
-:class:`~virtual_rainforest.core.model.BaseModel`. The existence of this
-:func:`~virtual_rainforest.core.model.BaseModel.__init_subclass__` function means that
-every child class is automatically added to the model registry (called
-:attr:`~virtual_rainforest.core.model.MODEL_REGISTRY`), but that in every case a
-``model_name`` has to be provided to register it under. This model name should be
-unique. Although registration is automatic, it only happens when class definition
-happens, which requires the module which defines the child class to be imported
-somewhere. At the moment, we import all child models in the top level ``__init__.py`` to
-ensure that they are added to the registry.
-
-An example of ``Model`` inheritance from
-:class:`~virtual_rainforest.core.model.BaseModel` can been seen in the :mod:`soil.model`
-module.
+"""The :mod:`~virtual_rainforest.core.model` module defines the high level API for the
+different models within the Virtual Rainforest. The module creates the
```

Comment:
for consistency, should we call this `core_model`?

Reply:
I think it's probably better to call it something like `model_api`. The reason I didn't update the name here to match the other model naming scheme is that it isn't really a model that would be used in the same way as other models, which giving it a consistent name could implicitly suggest that it was

Reply:
I think giving it a less ambiguous name would be good. I'd lean towards `base_model` - that seems to me clearer that it is the base for real models (and the object is called BaseModel!). `base_abc_model` would be even clearer but seems a bit over the top.

Perhaps make this a separate PR?

Reply:
Side note: once we have an agreement on this, could you please update the API links to the updated model names for soil and abiotic? We missed that when updating the model names.

Reply:
Yeah I would be happy with `base_model.py`